### PR TITLE
Darien Classrooms & Assignments Mk2 - Part 2

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview
This PR plans to improve the WildCam Darien Lab's Classrooms and Assignments systems, following up from Part 1 in #107 . I have an eye on making a generalised 'WildCam' Classroom & Assignment system (as opposed to being Darien-specific) but we'll see how that generalisation goes.

Goals:
- [x] Educators can create & edit Darien-specific Classrooms.
- [x] Educators can create & edit Darien-specific Assignments.
- [ ] [major] Educators can select Subjects from the Map Explorer to create Assignments.
- [ ] [code][long-term] 

Expected BIG STUFF that needs handling:
- Darien-specific Classroom & Assignment management will run head-long into the points discussed in #85 , since I'll need to address the hardcoded I2A things in the common Classrooms components.

NOT in the scope:
- Students pages. (i.e. "view my Assignments, view my Classrooms")

### Status
WIP

### CODE REVIEW

**Comment by shaunanoordin** [comment]
## PR Update
I'm stopping development on this branch so it can be properly reviewed; of note, I'd seriously appreciate the reorganisation of components having a look-over. 👍 

There's only one outstanding feature that will be spun off into its own PR:
- [major] Educators can select Subjects from the Map Explorer to create Assignments.

That feature was a bit buggy because I was trying something smart (i.e. dumb) with the storage of Subject selection, so I'll resort to something simpler later.

Also, Minor Weirdness:
- You can't add a new Darien Assignment after you created one. That's just a bad copy-paste that didn't account for different class logic.
- You'll notice some strange naming conventions, (some compos called `wildcamdarien`, others just `wildcam`) as I was originally trying to spin off several of the WildCam Darien components into generic WildCam components (i.e. ready for WildCam Gorongosa Lab to roll in) but there were some issues, e.g. the Join Link being hardcoded.

### Status
Ready for review, with further features to be taken care of later.

File: src/components/wildcam-classrooms/WildCamClassroomEditor.jsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+};
+
+WildCamClassroomEditor.defaultProps = {
+  editClassroom: () => {},
+  exportStats: () => {},
+  selectClassroom: () => {},
+  removeStudentFromClassroom: () => {},
+  //----------------
+  showConfirmationDialog: false,
+  showCounts: {
```
Comment (srallen) [inline src/components/wildcam-classrooms/WildCamClassroomEditor.jsx]: This probably doesn't apply?

File: src/components/wildcam-classrooms/WildCamClassroomEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+};
+
+WildCamClassroomEditor.propTypes = {
+  editClassroom: PropTypes.func,
+  exportStats: PropTypes.func,
+  selectClassroom: PropTypes.func,
+  removeStudentFromClassroom: PropTypes.func,
+  //----------------
+  showConfirmationDialog: PropTypes.bool,
+  showCounts: PropTypes.shape({
```
Comment (srallen) [inline src/components/wildcam-classrooms/WildCamClassroomEditor.jsx]: This probably doesn't apply for Wildcam.

File: src/containers/wildcam-classrooms/WildCamClassroomFormContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            id: this.props.selectedProgram.id,
+            type: 'programs'
+          }
+        }
+      }
+    };
+
+    return Actions.createClassroom(classroomData)
+      .then((classroom) => {
+        if (!this.props.selectedProgram.custom) {
```
Comment (srallen) [inline src/containers/wildcam-classrooms/WildCamClassroomFormContainer.jsx]: This only applies to I2A (a program with `custom: true` set). This is non-blocking comment and in my overall PR review, I'll go into the shared API request needs more.

**Comment by shaunanoordin** [comment]
## PR Update
This PR has been made obsolete by the new updates made in PR #122, #123, #124, and #125. These newer updates separate the WildCam components into their own independent modules, which are easier to extend into different WildCam classrooms (e.g. WildCam Darien, WildCam Gorongosa, etc)

**Pull request closed without merging**